src/interpreter/executor/interpret.js

```diff
@@ -0,0 +1,332 @@
+"use strict";
+// This is our 'state' when interpreting the code
+//  Use this enum as a flag for certain logic
+const stateEnum = {
+    DEFAULT : "default",
+    ACCEPTING_VAR : "accepting variable",
+    ASSIGNING_VAR : "assigning variable"
+}
+
+function Variable() {
```

Comment:
Is this a class? Try using the ES6 class syntax for clarity. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes